user/db.go

```diff
@@ -73,161 +70,276 @@ func (s *Store) updateUser(ctx context.Context, user User) error {
-func (s *Store) getUsersLLA(ctx context.Context) ([]User, error) {
+// getUsersSearchOptionsDesc will get users search with sorting desc
+func (s *Store) getUsersSearchOptionsDesc(ctx context.Context, search, sortBy string) ([]User, error) {
 	var u []User
 	err := s.db.SelectContext(ctx, &u,
-		`SELECT user_id, username, nickname, first_name, last_name, email, last_login, avatar, use_gravatar
+		`SELECT *
 		FROM people.users
-		ORDER BY last_login ASC NULLS FIRST;`)
+		WHERE (CAST(user_id AS TEXT) LIKE '%' || $1 || '%')
+		   OR (username LIKE '%' || $1 || '%')
+		   OR (nickname LIKE '%' || $1 || '%')
+		   OR (first_name LIKE '%' || $1 || '%')
+		   or (last_name LIKE '%' || $1 || '%')
+		   OR (email LIKE '%' || $1 || '%')
+		   OR (first_name || ' ' || last_name LIKE '%' || $1 || '%')
+		ORDER BY
+		    CASE WHEN $2 = 'userId' THEN user_id END DESC,
+		    CASE WHEN $2 = 'name' THEN first_name END DESC,
+			CASE WHEN $2 = 'name' THEN last_name END DESC,
+		    CASE WHEN $2 = 'username' THEN username END DESC,
+		    CASE WHEN $2 = 'email' THEN email END DESC,
+		    CASE WHEN $2 = 'lastLogin' THEN last_login END DESC NULLS LAST;`, search, sortBy)
 	if err != nil {
 		return nil, err
 	}
 	return u, nil
 }
 
-// getUsersLLD will get a group of users by lastLogin DESC
-func (s *Store) getUsersLLD(ctx context.Context) ([]User, error) {
+// getUsersSearchOptionsDescSizePage will get users search with sorting desc with size and page
+func (s *Store) getUsersSearchOptionsDescSizePage(ctx context.Context, search, sortBy string, size, page int) ([]User, error) {
 	var u []User
 	err := s.db.SelectContext(ctx, &u,
-		`SELECT user_id, username, nickname, first_name, last_name, email, last_login, avatar, use_gravatar
+		`SELECT *
 		FROM people.users
-		ORDER BY last_login DESC NULLS LAST;`)
+		WHERE (CAST(user_id AS TEXT) LIKE '%' || $1 || '%')
+		   OR (username LIKE '%' || $1 || '%')
+		   OR (nickname LIKE '%' || $1 || '%')
+		   OR (first_name LIKE '%' || $1 || '%')
+		   or (last_name LIKE '%' || $1 || '%')
+		   OR (email LIKE '%' || $1 || '%')
+		   OR (first_name || ' ' || last_name LIKE '%' || $1 || '%')
+		ORDER BY
+		    CASE WHEN $2 = 'userId' THEN user_id END DESC,
+		    CASE WHEN $2 = 'name' THEN first_name END DESC,
+			CASE WHEN $2 = 'name' THEN last_name END DESC,
+		    CASE WHEN $2 = 'username' THEN username END DESC,
+		    CASE WHEN $2 = 'email' THEN email END DESC,
+		    CASE WHEN $2 = 'lastLogin' THEN last_login END DESC NULLS LAST
+		LIMIT $3
+		OFFSET $4;`, search, sortBy, size, size*(page-1))
 	if err != nil {
 		return nil, err
 	}
 	return u, nil
 }
 
 // getPermissionsForUser returns all permissions for a user
-func (s *Store) getPermissionsForUser(ctx context.Context, u User) (p []permission.Permission, err error) {
-	err = s.db.SelectContext(ctx, &p, `SELECT p.permission_id, p.name, p.description
+func (s *Store) getPermissionsForUser(ctx context.Context, u User) (p []string, err error) {
```

Comment:
Why this change from `Permission` to `string`?

Reply:
I think at the time it was because of it's use case, it was only the name that was used but is fixed in a later version later on